plugins/woocommerce-admin/client/products/fields/attribute-input-field/attribute-input-field.tsx

```diff
@@ -46,41 +50,46 @@ export const AttributeInputField: React.FC< AttributeInputFieldProps > = ( {
 	} );
 
 	const getFilteredItems = (
-		allItems: Pick< QueryProductAttribute, 'id' | 'name' >[],
+		allItems: NarrowedQueryAttribute[],
 		inputValue: string
 	) => {
-		return allItems.filter(
-			( item ) =>
-				filteredAttributeIds.indexOf( item.id ) < 0 &&
+		const onlyIdsFilter = ( item: NarrowedQueryAttribute ) =>
+			onlyAttributeIds.length
+				? onlyAttributeIds.includes( item.id )
+				: true;
+
+		const ignoreIdsFilter = ( item: NarrowedQueryAttribute ) =>
+			ignoredAttributeIds.length
+				? ! ignoredAttributeIds.includes( item.id )
+				: true;
+
+		return allItems
+			.filter( onlyIdsFilter )
+			.filter( ignoreIdsFilter )
+			.filter( ( item ) =>
```

Comment:
This seems like a pretty efficient way to filter, could we combine all three of these into a single `filter` function?

Reply:
I wondered if someone would call me on that 😅. 

I've Googled around about this before, and my understanding is that unless we're dealing with thousands of items, the performance hit is completely negligible, and I personally find it much more readable this way. These built-in functions are generally optimized enough that I favor readability over optimization when it doesn't seem that it would make a real-world difference. 

I could refactor this into separately named booleans in a single filter if we wish, although I still find the chaining slightly more "clear" as a developer reading this.

Reply:
> But I don't think readability should be in favour of performance.

I would tend to disagree with that in general if the performance impact is negligible, but we no longer need this condition and it's relatively easy to compromise here 👍🏻